src/main/java/com/example/bookshop/repository/impl/BookRepositoryImpl.java

```diff
@@ -51,4 +49,14 @@ public List<Book> findAll() {
             throw new DataProcessingException("Can't get books ", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            Book book = entityManager.find(Book.class, id);
+            return Optional.ofNullable(book);
+        } catch (RuntimeException e) {
```

Comment:
```suggestion
        } catch (Exception e) {
```
Just to be sure we won't miss anything